### DISCUSSION TITLE
bicep-langserver installation help

### DISCUSSION THREAD

**Discussion by mahalel** [helix-editor/helix#7881] [Q&A] [answered]
Hey, I love this editor and thank you for a great project.
In the output of `hx --health` I can see there is support for bicep with `bicep-langserver`, however I cannot figure out how to get this installed so that Helix detects it and uses it correctly.

I have downloaded the zip for the [langserver](https://github.com/Azure/bicep/releases/latest/download/bicep-langserver.zip) and unzipped it in my $PATH (under /usr/local/bin/bicep-langserver). There are a bunch of `.dll` files in that folder but not so much an executable binary that I can see.

However, it is still not getting picked up by Helix, I wonder if anyone has got this working which could point me in the right direction please.

**Comment by woojiq** [comment]
Code responsible for detecting binarie:  https://github.com/helix-editor/helix/blob/48eb0d47925586a9dadea577fad607776631b4ab/helix-term/src/health.rs#L185-L189

Try running `which bicep-langserver` on your device, what's the result?
~~Make sure PATH doesn't contain a tilde, see: https://github.com/helix-editor/helix/issues/7266~~

**Reply by mahalel** [reply]
Hi, thanks for your answer `which: no bicep-langserver in ......paths` 
I don't think `bicep-langserver` has a linux executable per-se, it's literally a zip file which if you unzip is a bunch of dlls. I suspect it's a Windows thing only?

I'm only asking because it's showing up on the list of supported LSPs so I wondered if anyone got it working on linux/wsl2, as I'm not smart enough to figure it out.

**Reply by woojiq** [reply]
The zip archive has `Bicep.LangServer.exe`, so I assume it's built for Windows. Anyway, I think it's better to ask this question in the biceps repo than here. Or ask @lunjon who added this language to helix.
> I have a working configuration for the LSP that I use in Helix already. I could also write on how to install/setup the language server.

**Reply by lunjon** [reply, marked as answer]
Hi! Sorry for late reply (I'm on parental leave).

You are correct in everything you say, and there's no executable for the language server. However, I have followed a convention used in several other language servers and created a script file that is executable and put in a directory somewhere in `$PATH`:

```sh
#!/usr/bin/env bash
exec dotnet /home/myuser/.cache/bicep-langserver/Bicep.LangServer.dll
```

I'm not a Windows user myself so I don't know how to solve it other than overriding command in the configuration.

**Reply by mahalel** [reply]
@lunjon that is beautiful, that works perfectly for me now! @woojiq I don't believe I can contribute directly to the wiki to add this information in. What is the best process to get the wiki updated for bicep please?

**Reply by gabydd** [reply]
You should be able to edit the wiki directly there should be an edit button in the top right corner

**Reply by mahalel** [reply]
Thanks @gabydd I didn't realise the permissions were so open. I have added the instructions now - https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#bicep

Appreciate your help all :)
